### DISCUSSION TITLE
Understanding charge strategies

### DISCUSSION THREAD

**Discussion by alastaid** [firstof9/openevse#240] [General]
Perhaps I have missed some docs somewhere, but I am struggling a bit trying to get HA to control my OpenEVSE.  I was  using the OpenEVSE schedule 2-5am to use cheap electricity, and now want to use HA to achieve the same, as I want more flexibility for other strategies.  I used the switch.openevse_sleep_mode to turn off at 2am, vehicle starts charging, turn the same switch on at 5am, but this doesnt put the openevse into sleep and keeps charging until the car is fully charged.  What should I do to stop the charging at 5am?  Thanks in advance

**Comment by alastaid** [comment]
So spent a while reading the API calls etc, and PR's, to fully control the OpenEVSE from HA, should I be using the openevse.set_override with state: enabled & state:disabled for starting and stopping charging?  As HA controls all, I would not need to use openevse.clear_override ?   Lastly, what should I call when I want to use solar divert mode, do I only need to use switch.openevse_solar_pv_divert and not use openevse.clear_override in this scenario?  Thanks

**Reply by firstof9** [reply]
`openevse.set_override` with `state: enabled` means charge, `state: disabled` means no charge.

Using `openevse.clear_override` would technically be the same as `state: enabled` but it will also clear any amperage override you have in place as well.

The `switch.openevse_solar_pv_divert` enables the PV divert/Self Production setting in the OpenEVSE, this function seems to be hardwired to require a MQTT topic in order to function correctly.

**Reply by alastaid** [reply]
Thanks so much for the reply, with regard to the solar divert and the MQTT topic, are you referring to the topic you enter in the settings page which points to the grid export/import value or something different?  Would you also expect to set the OpenEVSE to openevse.set_override enabled or disabled before setting the solar_divert mode.  Apologies for these questions, but I am trying to get it right without keep tripping on the OpenEVSE and my car with short charging events trying to get the correct behaviour, which I keep doing at the moment.

**Reply by firstof9** [reply]
`openevse.set_override` is completely seperate, it all depends on how you want to do your charging.

I have mine setup with a schedule in the OpenEVSE to only allow charging on off peak hours, this is stand alone from HA. I however don't utilize any PV divert.

> are you referring to the topic you enter in the settings page which points to the grid export/import value or something different?

Yes that's the setting.
